src/vendor/processors/stripe.py

```diff
@@ -1402,49 +1409,34 @@ def get_subscription(self, stripe_subscription):
-        it is recommended to use the checkout flow.
-
-        It is expected that before calling this method the `set_billing_address_form_data` and `set_payment_info_form_data`
-        have been called.
-
+    def create_subscription(self, payment_method_id, subscription_extras):
+        """Creates a Stripe and Vendor Subscriptions instance
+        
+        Method setups the process to create and charge a Stripe Subscription. If a Stripe Subscription
+        object is returned a Vendor Subscrption instance will be created. Otherwise an excepetion
+        will be raise with the reason of why the subscription failed to create.
+        
         Args:
-            offer: Offer instance that will be added to an invoice to create the subscription
-
-            previous_start_date: DateTime instance. Helps to calcualate when the next billing cylce will start for
-            the subscription. If None it will start when the code is executed.
+            payment_method_id: String coming from a Stripe Payment Method Object
+            
+            subscription_extras: Dictionary containing extra arguments for the Stripe Subscription.
+            EG { "trail_days": 10, "billing_cycle_anchor": <datetime>}
         
         Returns:
-            Subscription Instance
-
+            subscription: Vendor Subscription Model instance
+            
         Exceptions:
-            Stripe Payment Method.
-
-            Stripe Setup Intent.
-
-            Subscription Failed to be created.
-
+            Stripe Setup Intent failed to create
+            
+            Stripe Subscription failed to create
         """
-        payment_method_data = self.build_payment_method()
-        stripe_payment_method = self.stripe_create_object(self.stripe.PaymentMethod, payment_method_data)
-        if not stripe_payment_method:
-            raise Exception(f"Could not create stripe_payment_method transaction_info: {self.transaction_info}")
-
-        setup_intent_object = self.build_setup_intent(stripe_payment_method.id)
+        setup_intent_object = self.build_setup_intent(payment_method_id)
         stripe_setup_intent = self.stripe_create_object(self.stripe.SetupIntent, setup_intent_object)
         if not stripe_setup_intent:
             raise Exception(f"Could not create stripe_setup_intent transaction_info: {self.transaction_info}")
-
-        subscription_obj = self.build_subscription(self.invoice.order_items.first(), stripe_payment_method.id)
-        if previous_start_date:
-            trial_days = offer.get_offer_end_date(previous_start_date) - timezone.now()
-            subscription_obj['trial_period_days'] = trial_days.days
-            subscription_obj['billing_cycle_anchor'] = offer.get_offer_end_date(previous_start_date)
+        
+        subscription_obj = self.build_subscription(self.invoice.order_items.first(), payment_method_id)
+        subscription_obj |= subscription_extras
```

Comment:
not sure what's going on here- is this supposed to add the subscription extras to the object? 
from what i can tell this line equals `subscription_obj = subscription_obj or subscription_extras`

Reply:
It's a handy way merge two dictionarys